File: Beecrowd_Parte4/Presente_1425.java
import java.util.LinkedList;
import java.util.Queue;
import java.util.Scanner;

public class Presente_1425 {

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        while (true) {
            int N = scanner.nextInt();
            int M = scanner.nextInt();
            if (N == 0 && M == 0) break;

            if (canReachStone(N, M)) {
                System.out.println("Let me try!");
            } else {
                System.out.println("Don't make fun of me!");
            }
        }
    }

    private static boolean canReachStone(int N, int M) {
        // BFS setup
        Queue<Position> queue = new LinkedList<>();
        boolean[] visited = new boolean[N + 2]; // Extra space for left and right bank

        // Start from the left bank (position 0)
        queue.add(new Position(0, 1));
        visited[0] = true;

        while (!queue.isEmpty()) {
            Position current = queue.poll();
            int currentPosition = current.position;
            int jumpNumber = current.jumpNumber;

            // Calculate the next jump distance
            int jumpDistance = 2 * jumpNumber - 1;

            // Jump forward
            int forwardPosition = currentPosition + jumpDistance;
            if (forwardPosition == M) return true; // Reached the target stone
            if (forwardPosition <= N && !visited[forwardPosition]) {
                visited[forwardPosition] = true;
                queue.add(new Position(forwardPosition, jumpNumber + 1));
            }

            // Jump backward
            int backwardPosition = currentPosition - jumpDistance;
            if (backwardPosition == M) return true; // Reached the target stone
            if (backwardPosition > 0 && !visited[backwardPosition]) {
                visited[backwardPosition] = true;
                queue.add(new Position(backwardPosition, jumpNumber + 1));
            }
        }

        return false;
    }

    // Helper class to store the position and the jump number
    static class Position {
        int position;
        int jumpNumber;

        Position(int position, int jumpNumber) {
            this.position = position;
            this.jumpNumber = jumpNumber;
        }
    }
}

File: Beecrowd_Parte4/Presente_1425.c
#include <stdio.h>
#include <stdlib.h>
#include <stdbool.h>

#define MAX_N 1000000

typedef struct {
    int position;
    int jumpNumber;
} Position;

typedef struct {
    Position *data;
    int front;
    int rear;
    int size;
} Queue;

Queue* createQueue(int size) {
    Queue* queue = (Queue*)malloc(sizeof(Queue));
    queue->data = (Position*)malloc(size * sizeof(Position));
    queue->front = 0;
    queue->rear = 0;
    queue->size = size;
    return queue;
}

void enqueue(Queue* queue, Position value) {
    queue->data[queue->rear++] = value;
}

Position dequeue(Queue* queue) {
    return queue->data[queue->front++];
}

bool isEmpty(Queue* queue) {
    return queue->front == queue->rear;
}

void freeQueue(Queue* queue) {
    free(queue->data);
    free(queue);
}

bool canReachStone(int N, int M) {
    // BFS setup
    Queue* queue = createQueue(N + 2);
    bool *visited = (bool *)calloc(N + 2, sizeof(bool));

    // Start from the left bank (position 0)
    Position start = {0, 1};
    enqueue(queue, start);
    visited[0] = true;

    while (!isEmpty(queue)) {
        Position current = dequeue(queue);
        int currentPosition = current.position;
        int jumpNumber = current.jumpNumber;

        // Calculate the next jump distance
        int jumpDistance = 2 * jumpNumber - 1;

        // Jump forward
        int forwardPosition = currentPosition + jumpDistance;
        if (forwardPosition == M) {
            freeQueue(queue);
            free(visited);
            return true; // Reached the target stone
        }
        if (forwardPosition <= N && !visited[forwardPosition]) {
            visited[forwardPosition] = true;
            Position next = {forwardPosition, jumpNumber + 1};
            enqueue(queue, next);
        }

        // Jump backward
        int backwardPosition = currentPosition - jumpDistance;
        if (backwardPosition == M) {
            freeQueue(queue);
            free(visited);
            return true; // Reached the target stone
        }
        if (backwardPosition > 0 && !visited[backwardPosition]) {
            visited[backwardPosition] = true;
            Position next = {backwardPosition, jumpNumber + 1};
            enqueue(queue, next);
        }
    }

    freeQueue(queue);
    free(visited);
    return false;
}

int main() {
    int N, M;
    while (scanf("%d %d", &N, &M) == 2) {
        if (N == 0 && M == 0) break;

        if (canReachStone(N, M)) {
            printf("Let me try!\n");
        } else {
            printf("Don't make fun of me!\n");
        }
    }
    return 0;
}
